docs/developing/troubleshooting.md

```diff
@@ -63,3 +63,27 @@ If your app wants to write to locations other than `/var`, and it's not easy to
 change the app's code, one work-around is to create a symlink from the location
 you app wants to modify to a location under `/var`.  This won't work for all
 applications however.
+
+## ETag data
+
+Sandstorm drops invalid ETag data. This typically has no impact on app functionality. When Sandstorm
+drops invalid ETag data, it stores a warning in the grain log that it did so, limiting the log
+messages to at most one per launch of a grain to avoid filling up the log file. See also [issue
+#2295.](https://github.com/sandstorm-io/sandstorm/issues/2295)
+
+In versions v0.176 and earlier, the grain would crash, rather than log an error message, if an
```

Comment:
My main purpose for this docs section existing is troubleshooting an app. If ocdtrekkie were to visit this page, and didn't really know what ETags are, and came here via a link that was emitted by sandstorm-http-bridge, I want him to have enough information to understand why the app's ETag header is problematic, and have enough info to file a bug against the app to have them fix the app. See also the docs URL that I added to `sandstorm-http-bridge` to provide "information scent" to allow app authors to understand and possibly fix the problem.

Proposed new text for the section:

## ETag data

Web apps often use the `ETag` HTTP response header to control caching. In Sandstorm, this header is
permitted so long as it follows the HTTP specification's requirement that the header values are
formatted with double-quote marks, like: `ETag: "value"` or `ETag: W/"value"`.

Some apps use strings without quotation marks in their ETag headers, such as `ETag: value`, which
violates the HTTP standard.  Sandstorm does not permit invalid header values because ambiguity can
lead to security problems. Therefore, Sandstorm will drop invalid `ETag` headers and will write a
warning to the grain's debug log. Usually, this does not affect app functionality, other than to
reduce the effectiveness of the browser's cache.

The headers are dropped within the grain at the
[sandstorm-http-bridge](../using/how-it-works.md) layer. Recent versions of `sandstorm-http-bridge` drop the header and
log a message. In versions v0.176 and earlier of `sandstorm-http-bridge`, invalid ETag data would cause the grain to crash rather than log an error message.